src/test/scala/advancepassengerinfo/importer/processor/DqFileProcessorImplTest.scala

```diff
@@ -209,6 +210,68 @@ class DqFileProcessorTest extends TestKit(ActorSystem("MySpec"))
       probe.expectMsg(ManifestCall(jsonFileName, manifest))
       probe.expectMsg(JsonFileCall("c", jsonFileName, successful = true, dateIsSuspicious = false))
       probe.expectMsg(ZipFileCall("c", successful = true))
+
+    }
+
+    "handle exceptions while getting nextFiles" in {
+      val mockFileNamesProvider = new FileNames {
+        override val nextFiles: String => Future[List[String]] = (previous: String) => Future.sequence(List(
+          Future.successful(List("1.zip", "2.zip")),
+          Future.failed(new Exception("next file exception")),
+          Future.successful(List("3.zip", "4.zip")),
+        )).map(_.flatten)
+      }
+
+      val probe = TestProbe("probe")
+      val mockPersistence = MockPersistence(probe.ref)
+
+      val manifests = Seq(
+        ("1.json", Success(manifest)),
+        ("2.json", Success(manifest)),
+      )
+      val processor = DqFileProcessorImpl(singleZipMockProvider(manifests), mockPersistence)
+      val dqApiFeed: DqApiFeedImpl = DqApiFeedImpl(
+        mockFileNamesProvider,
+        processor,
+        100.millis,
+        MockStatsDCollector
+      )
+
+      val result = Await.result(dqApiFeed.processFilesAfter("_").runWith(Sink.seq),1.second)
+
+      result should ===(Vector.empty)
```

Comment:
Why do we expect the result to be empty? Shouldn't it be returning the 2 successfully processed files?